.github/workflows/test.yml

```diff
@@ -0,0 +1,98 @@
+# This workflow will build a Java project with Gradle
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-gradle
+
+name: Run Tests
+
+on:
+  push:
+    branches: [ dev ]
+  pull_request:
+    branches: [ '**' ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
Pulling my earlier comment into the appropriate code location:

I'd love if you could comment above the strategy.matrix section with some notes summarizing the situation here, ideally with a breadcrumb or two for someone who is wondering what's going on. Maybe a link to https://github.community/t/support-for-yaml-anchors/16128/46 as the best "this would work if it existed" feature request.